app/safe/cmd/main.go

```diff
@@ -15,12 +15,20 @@ import (
 
 	"github.com/vmware-tanzu/secrets-manager/app/safe/internal/bootstrap"
 	"github.com/vmware-tanzu/secrets-manager/app/safe/internal/server"
-	"github.com/vmware-tanzu/secrets-manager/core/log"
+	"github.com/vmware-tanzu/secrets-manager/core/crypto"
+	log "github.com/vmware-tanzu/secrets-manager/core/log/std"
 	"github.com/vmware-tanzu/secrets-manager/core/probe"
 )
 
 func main() {
-	id := "VSECMSAFE"
+	id, err := crypto.RandomString(8)
+	if err != nil {
+		id = "VSECMSAFE"
+	}
+
+	ctx, cancel := context.WithCancel(
+		context.WithValue(context.Background(), "correlationId", &id),
```

Comment:
Passing the correlation id as soon as the execution chain starts, and reusing it via the context.